core/src/main/java/org/apache/iceberg/BaseRowDelta.java

```diff
@@ -45,4 +62,80 @@ public RowDelta addDeletes(DeleteFile deletes) {
     add(deletes);
     return this;
   }
+
+  @Override
+  public RowDelta validateFromSnapshot(long snapshotId) {
+    this.startingSnapshotId = snapshotId;
+    return this;
+  }
+
+  @Override
+  public RowDelta validateDeletedFiles() {
+    return validateDeletedFiles(true);
+  }
+
+  public RowDelta validateDeletedFiles(boolean shouldValidate) {
```

Comment:
Is this public on purpose?

Reply:
No. I was debating whether to add it to the public API, so it was initially public. But I decided to keep the public API minimal for now. We can always add it later. That's why I kept it here.

Reply:
I'm just going to remove this. We get warnings if it is private.